app/src/main/java/com/kickstarter/ui/viewholders/projectcampaign/AudioElementViewHolder.kt

```diff
@@ -93,6 +86,7 @@ class AudioElementViewHolder(
 
     override fun destroy() {
         mediaPlayer.release()
+        disposables.dispose()
```

Comment:
@Arkariang Is this the right place to dispose of the disposables? I'm not too confident about how KSViewHolder works. Also, why is super.destroy() not called first in this method?

Reply:
You're right! this `on destroy` works in combination with `KSListAdapter: onViewDetachedFromWindow(holder: KSViewHolder)` -> `KSViewHolder.java:70`

But in this case the adapter is `ViewElementAdapter`, we should add on `ViewElementAdapter#onViewDetachedFromWindow` the destroy call

Reply:
Oof that's confusing, let me try to understand: you are saying this `AudioElementViewHolder#destroy` method is called by KSListAdapter. But `AudioElementViewHolder`'s adapter is **NOT** `KSListAdapter`, it is `ViewElementAdapter`. 

Then would `AudioElementViewHolder#destroy` ever get called?

Reply:
Correct, in the current implementation  AudioElementViewHolder#destroy will never be called